modyn/trainer_server/internal/trainer/remote_downsamplers/abstract_matrix_downsampling_strategy.py

```diff
@@ -120,3 +120,8 @@ def init_downsampler(self) -> None:
     def _select_indexes_from_matrix(self, matrix: np.ndarray, target_size: int) -> tuple[list[int], torch.Tensor]:
         # Here is where the actual selection happens
         raise NotImplementedError()
+
+    @property
+    def requires_grad(self) -> bool:
+        # Default to true if None
+        return self.matrix_content is None or self.matrix_content == MatrixContent.GRADIENTS
```

Comment:
`self.matrix_content` will be initialized by the subclasses. We can add an assertion `assert self.matrix_content is not None` here. In the actual return statement then we don't bother with `None`. See the `inform_samples` method in this class for example.

Or (actually a better way) we can add another argument `matrix_content` to the constructor of this baseclass, passed by the subclass to initialize the `self.matrix_content`. This way we can completely get rid of the nullity.

Reply:
the None case is True for tests, though. I needed to added it for unit test cases. Let me take another look

Reply:
moving to constructor